NatLibFi/melinda-rest-api-http
# Features: Record-import to Rest pipeline // merge // bulk-logs (#26)

* addrecord, get and setstate to bulk and some middleware updates

* new bulk routes

* update create bulk logic

* return correlation id on no stream bulk creation

* change create bulk response

* fix content-types prio

* fix query states params

* fix updateState

* log added record

* fix added record

* swap post routes

* body parser for record

* send bulk added record to pending_validation.correlationId

* object parametter to queryById

* add prio to headers of prio requests

*  Add mongoLog to bulk

    * Add mongoOperator for logPrio -collection to prio
    * Accept status MERGED if for records if merge -param is true
    * Get doQuery -results from Mongo logPrio instead of actual Mongo

*  Check that merge is not used with unique:0
* Add merge to apidoc
* Add merge-parameter for prio jobs

* Use operationSettings param object

* make amqp headers in route when adding record

* drone to github actions (#31)
* remove drone yml 
* add github workflow
* add dependabot yml
* use 2.2.0-alpha commons

* Updates to batchBulk

* Update contentType and accept checks

* Update CONTENT_TYPE constant

* Update CONVERSION_FORMATS constant

* fix remove bulk

* bulk utils and patch audit

* add bulk query check

* better response

* add x-powered-bu disable

* Add blobSequence to added records

* Updates to response handling

* Set httpStatus CREATED for a succesfull priocreate

* Update error handling

* recordResponseItem: melindaId -> databaseId

* Error for invalid bulk validate/unique/merge param

* improve bulk query

* fix querys

* fix timestamp validation

* implement skip and limit in querys

* use query check in prio too + implement query search methods from bulk

* add files

* fix prio query

* Fixes to operationSettings

* add query params for limit returning stuff

* split params checks

* better errors for readContent

* rejected file and log file parametters from 50 char to 100

* rejected file and log file parametters allow _ and fix regexp

* Fix getState for non-existent jobs

* Add logs functionality

* Fix using blobSequence params in LogsQuery

* Add recordsAsReport parameter to bulks doQuery

* Check validity of id for prio UPDATEs

Valid Aleph-IDs are 9 character long and include only digits.
- Check that id given in query is a valid Aleph-ID for prio UPDATEs

* Update deps

* Update githubActions

* add license-scan
* add node-version 18x check

* Fix gitHubActions

* Add gitHubActions CODEOWNERS

* Update GitHubActions license-scan

* Reformat log messages

* Add noRecords param to recordsAsReport

* Check recordReport parameters

* Add noIds parameter

* update and add limit & skip to logs

* improve log query

* improve log query debug

* new route: logs

* fix protect

* Accept true false as parameters (#37)

* Update melinda-commons
* Recognize true and false as valid, fix regex

* Update melinda-rest-api-commons (#38)

* Skip no change updates (#36)

* Add new status SKIPPED (http 200)
* Add new parameter skipNoChangeUpdates

* Update melinda-rest-api-commons
* Update melinda-commons

* Use recordStatus as status

* Add recordType parameter (#39)

* Add recordType parameter ** settable in env RECORD_TYPE
** defaults to 'bib'

* Error using unique or merge for creates/updates if recordType is something else than 'bib'

* Update apidoc for prio and bulk (#40)

* Update apidoc for prio
* Add 'new' bulk operations to api.json

* Update deps (#41)

* Update GitHub actions (#44)

* Add openshift-webhook

* Update openshift-webhook branch

* Update melinda rest api commons (#45)

* Update melinda-rest-api-commons

* Use JSON.stringify instead of inspect in log (#47)

* Add webhook for auth prod

* Update deps (#49)

Co-authored-by: Minttu Hurme <ammsalme@users.noreply.github.com>
